ta4j-core/src/main/java/org/ta4j/core/indicators/supertrend/SuperTrendIndicator.java

```diff
@@ -0,0 +1,94 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.supertrend;
+
+import org.ta4j.core.Bar;
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.indicators.ATRIndicator;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+import org.ta4j.core.num.DoubleNum;
+import org.ta4j.core.num.Num;
+
+public class SuperTrendIndicator extends RecursiveCachedIndicator<Num> {
+    private final UpperBandIndicator upperBandIndicator;
+    private final LowerBandIndicator lowerBandIndicator;
+
+    public SuperTrendIndicator(final BarSeries series, int length, final Integer multiplier) {
+        super(series);
+        ATRIndicator atrIndicator = new ATRIndicator(series, length);
+        this.upperBandIndicator = new UpperBandIndicator(series, atrIndicator, multiplier);
+        this.lowerBandIndicator = new LowerBandIndicator(series, atrIndicator, multiplier);
+    }
+
+    public SuperTrendIndicator(final BarSeries series) {
+        this(series, 3, 10);
+    }
+
+    @Override
+    protected Num calculate(int i) {
+        Num value = DoubleNum.valueOf(0);
+
+        if (i == 0)
+            return value;
+        Bar bar = getBarSeries().getBar(i);
+        /*
+         * if(bar.getClosePrice().isLessThanOrEqual(this.upperBandIndicator.getValue(i))
+         * ) value = this.upperBandIndicator.getValue(i); else value =
+         * this.lowerBandIndicator.getValue(i);
+         */
+
+        if (this.getValue(i - 1).isEqual(this.upperBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isLessThanOrEqual(this.upperBandIndicator.getValue(i))) {
+            value = this.upperBandIndicator.getValue(i);
+        }
+
+        if (this.getValue(i - 1).isEqual(this.upperBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isGreaterThan(this.upperBandIndicator.getValue(i))) {
+            value = this.lowerBandIndicator.getValue(i);
+        } else if (this.getValue(i - 1).isEqual(this.lowerBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isGreaterThanOrEqual(this.lowerBandIndicator.getValue(i))) {
+            value = this.lowerBandIndicator.getValue(i);
+        } else if (this.getValue(i - 1).isEqual(this.lowerBandIndicator.getValue(i - 1))
+                && bar.getClosePrice().isLessThan(this.upperBandIndicator.getValue(i))) {
+            value = this.upperBandIndicator.getValue(i);
+        }
+
+        return value;
+    }
+
+    public String getSignal(int index) {
+        if (getBarSeries().getBar(index).getClosePrice().isLessThanOrEqual(this.getValue(index))) {
+            return "Sell";
```

Comment:
Would it make sense to use the Trade.TradeType enum here instead of the string literals?

Reply:
Used Trade.TradeType as per your suggestion. Kept method return type as String.